ReactWindows/ReactNative/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -1,15 +1,34 @@
-﻿using ReactNative.Bridge;
+﻿using Newtonsoft.Json.Linq;
+using ReactNative.Bridge;
+using System;
+using Windows.Storage;
 
 namespace ReactNative.Modules.Storage
 {
     /// <summary>
     /// The asynchronous storage module.
     /// </summary>
-    /// <remarks>
-    /// TODO: implement.
-    /// </remarks>
-    public class AsyncStorageModule : NativeModuleBase
+    public class AsyncStorageModule : ReactContextNativeModuleBase
     {
+        private readonly string _arr = "A";
+        private readonly string _nul = "N";
+        private readonly string _str = "S";
+
+        private readonly string _invalidKey = "Invalid key";
+        private readonly string _invalidPair = "Invalid key value pair";
+
+        private ApplicationDataContainer _container; 
+
+        /// <summary>
+        /// Instantiates the <see cref="AsyncStorageModule"/>.
+        /// </summary>
+        /// <param name="reactContext">The context.</param>
+        internal AsyncStorageModule(ReactContext reactContext)
+            : base(reactContext)
+        {
+            _container = ApplicationData.Current.LocalSettings.CreateContainer(Name, ApplicationDataCreateDisposition.Always);
```

Comment:
The container is within the application's own local storage. When the app package is made is it possible that there are several independent React apps within it so that we need multiple storages?